tools/export/cdt/__init__.py

```diff
@@ -38,19 +38,23 @@ class Eclipse(Makefile):
     """Generic Eclipse project. Intended to be subclassed by classes that
     specify a type of Makefile.
     """
-    def get_target_config(self, ctx, configuration):
+    def get_target_config(self, configuration):
         """Retrieve info from cdt_definitions.json"""
-        tgt = deepcopy(TARGET_MAP[self.target])
-        defaults = deepcopy(_CONFIGS_OPTIONS['default'])
+        defaults = _CONFIGS_OPTIONS['default']
         eclipse_config = deepcopy(defaults['generic'])
         if configuration in defaults:
             eclipse_config.update(defaults[configuration])
 
+        # Get target-specific options. Use resolution order and extra labels
+        # to find alias for target. This allows to define options within inheritance path
         target_specific = _CONFIGS_OPTIONS['targets']
-        if tgt.name in target_specific:
-            eclipse_config.update(target_specific[tgt.name]['generic'])
-            if configuration in target_specific[tgt.name]:
-                eclipse_config.update(target_specific[tgt.name][configuration])
+        aliases = self.toolchain.target.resolution_order_names \
+            + self.toolchain.target.extra_labels
+        target_alias = next((t for t in aliases if t in target_specific), None)
+        if target_alias:
+            eclipse_config.update(target_specific[target_alias]['generic'])
+            if configuration in target_specific[target_alias]:
+                eclipse_config.update(target_specific[target_alias][configuration])
```

Comment:
I would write this differently, as I generally avoid using the `next` function:
```python
for target_alias in aliases:
    if target_alias in target_specific:
        eclipse_config.update(target_specific[target_alias]['generic'])
        if configuration in target_specific[target_alias]:
            eclipse_config.update(target_specific[target_alias][configuration])
        return eclipse_config
```

Reply:
Applied your fix, thanks.

Reply:
You dropped (or mis-indented) the return, which makes the code apply _all_ possible target specific settings, as opposed to what it used to do which is apply the _first_ match or most specific match.

Reply:
Missed that. I added a break instead of return, because we still have to return defaults, even if specific settings does not exist.